src/java/org/apache/cassandra/management/BaseCommand.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.management;
+
+import javax.inject.Inject;
+
+import org.apache.cassandra.tools.Output;
+
+/**
+ * Base class for all nodetool commands.
+ */
+public abstract class BaseCommand implements Runnable
+{
+    @Inject
+    protected Output logger;
+
+    /** The ServiceBridge instance to interact with the Cassandra node. */
+    protected ServiceMBeanBridge bridge;
+
+    /**
+     * The ServiceBridge instance is injected by the picocli framework during command execution and is used to
+     * interact with the Cassandra node. This method is called by picocli and used depending on the execution strategy.
+     * @param bridge The ServiceBridge instance to inject.
+     */
```

Comment:
Is the goal here to diverge from `NodeToolCmd` in the sense that `BaseCommand` is supposed to be decoupled from JMX?

Reply:
Yes, as discussed we'll move it to another issue.